src/mono/wasm/debugger/BrowserDebugProxy/ValueTypeClass.cs

```diff
@@ -53,48 +53,74 @@ public ValueTypeClass(byte[] buffer, string className, JArray fields, int typeId
                                                 int typeId,
                                                 int numValues,
                                                 bool isEnum,
+                                                bool includeStatic,
                                                 CancellationToken token)
         {
             var typeInfo = await sdbAgent.GetTypeInfo(typeId, token);
             var typeFieldsBrowsableInfo = typeInfo?.Info?.DebuggerBrowsableFields;
             var typePropertiesBrowsableInfo = typeInfo?.Info?.DebuggerBrowsableProperties;
 
             IReadOnlyList<FieldTypeClass> fieldTypes = await sdbAgent.GetTypeFields(typeId, token);
-            // statics should not be in valueType fields: CallFunctionOnTests.PropertyGettersTest
+
+            JArray fields = new();
+            if (includeStatic)
+            {
+                IEnumerable<FieldTypeClass> staticFields = fieldTypes
+                .Where(f => f.Attributes.HasFlag(FieldAttributes.Static));
+                foreach (var field in staticFields)
+                {
+                    var fieldValue = await sdbAgent.GetFieldValue(typeId, field.Id, token);
+                    fields.Add(GetFieldWithMetadata(field, fieldValue, isStatic: true));
+                }
+            }
+
             IEnumerable<FieldTypeClass> writableFields = fieldTypes
                 .Where(f => !f.Attributes.HasFlag(FieldAttributes.Literal)
                     && !f.Attributes.HasFlag(FieldAttributes.Static));
 
-            JArray fields = new();
             foreach (var field in writableFields)
             {
                 var fieldValue = await sdbAgent.ValueCreator.ReadAsVariableValue(cmdReader, field.Name, token, true, field.TypeId, false);
+                fields.Add(GetFieldWithMetadata(field, fieldValue, isStatic: false));
+            }
+
+            long endPos = cmdReader.BaseStream.Position;
+            cmdReader.BaseStream.Position = initialPos;
+            byte[] valueTypeBuffer = new byte[endPos - initialPos];
+            cmdReader.Read(valueTypeBuffer, 0, (int)(endPos - initialPos));
+            cmdReader.BaseStream.Position = endPos;
 
-                fieldValue["__section"] = field.Attributes switch
+            return new ValueTypeClass(valueTypeBuffer, className, fields, typeId, isEnum);
+
+            JObject GetFieldWithMetadata(FieldTypeClass field, JObject fieldValue, bool isStatic)
+            {
+                // GetFieldValue returns JObject without name and we need this information
+                if (isStatic)
+                    fieldValue["name"] = field.Name;
+
+                fieldValue["__section"] = isStatic
+                    ? field.Attributes switch
+                    {
+                        FieldAttributes.Private | FieldAttributes.Static => "private",
```

Comment:
Check with `HasFlag` instead of this, because this would be doing an exact match of the value. But the field attributes could have other flags too.

Reply:
When `attrs = Assembly`  and does not include Private, `attr.HasFlag(FieldAttributes.Private) == true`. To fix it I changed the order of arguments to check for Assembly instead of Private and leave Private as the 3rd option and it works but it's kind of strange.